SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -170,14 +168,15 @@ size_t get_index(size_t box, size_t row, size_t column){
 
 
 struct Pokemon{
-    std::vector<BoxSortingSelection>* preferences;
+    const std::vector<BoxSortingSelection>* preferences;
```

Comment:
I meant a real reference (everything that could be a reference instead of a ptr should be imo, that way you're sure it is never null when using it) :
const std::vector<BoxSortingSelection>&
but that is fine nonetheless

Reply:
Oh! I can do that!

Reply:
PokemonHome_BoxSorting.cpp:445: error: C2664: 'void std::swap(std::thread &,std::thread &) noexcept': cannot convert argument 1 from '_Ty' to 'std::thread &'

Seems swap is unhappy when I do that.

Reply:
oh true, my bad !